src/main/groovy/org/checkerframework/gradle/plugin/CreateManifestTask.groovy

```diff
@@ -0,0 +1,42 @@
+package org.checkerframework.gradle.plugin
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * This task generates a manifest file in the style of
+ * https://checkerframework.org/manual/#checker-auto-discovery
+ * Once the plugin places this directory containing it on the annotation processor path,
+ * javac can find the checkers via annotation processor discovery.
+ */
+class CreateManifestTask extends DefaultTask {
+
+    @Input
+    String[] checkers = []
+
+    /**
+     * Creates a manifest file listing all the checkers to run.
+     */
+    @TaskAction
+    def generateManifest() {
+        def manifestDir = project.mkdir "${project.buildDir}/checkerframework/META-INF/services/"
+        def manifestFile = project.file(manifestDir.absolutePath + "/javax.annotation.processing.Processor")
+        if (!manifestFile.createNewFile()) {
+            manifestFile.delete()
+            manifestFile.createNewFile()
+        }
+        manifestFile << this.checkers.join('\n')
+    }
+
+    /**
+     * Required so that this can incrementalized correctly, and so that `gradle clean build` behaves
+     * the same as `gradle clean ; gradle build`.
+     */
+    @OutputFile
+    def getManifestLocation() {
+        return "${project.buildDir}/checkerframework/META-INF/services/javax.annotation.processing.Processor"
```

Comment:
Minor:  there are two occurrences of the directory string, so you might consider abstracting that out.